cypress/utils/apps/query_enhancements/shared.js

```diff
@@ -46,18 +46,22 @@ export const getRandomizedDatasourceName = () =>
 /**
  * Returns an array of test configurations for every query language + dataset permutation
  * @param {GenerateTestConfigurationCallback} generateTestConfigurationCallback - cb function that generates a test case for the particular permutation
+ * @param {Object} [options] - Optional configuration options
+ * @param {string} [options.indexPattern] - Custom index pattern name (defaults to INDEX_PATTERN_WITH_TIME)
+ * @param {string} [options.index] - Custom index name (defaults to INDEX_WITH_TIME_1)
  * @returns {object[]}
  */
-export const generateAllTestConfigurations = (generateTestConfigurationCallback) => {
+export const generateAllTestConfigurations = (generateTestConfigurationCallback, options = {}) => {
+  const { indexPattern = INDEX_PATTERN_WITH_TIME, index = INDEX_WITH_TIME_1 } = options;
```

Comment:
I'm fine with this change, but looking at it's usage in this PR, it looks like the only change is:

```
indexPattern: 'data_logs_small_time_1*',
```

Can't we just change this to `data_logs_small_time_*`?

Reply:
But irreglardless of my questions, lets keep this change (i found an issue when migrating that'll make this useful!)

Reply:
Yeah I want it to be more general. So you can define what is index patten and index.